File: EduHubConnect-Backend-UsingSpringBoot/src/main/java/com/connect/eduHub/dto/PostDTO.java
package com.connect.eduHub.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.RequiredArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import com.connect.eduHub.model.Comment;
import com.connect.eduHub.model.Like;

@Data
@AllArgsConstructor
@RequiredArgsConstructor
public class PostDTO {
   
	private Long id;
    private String content;
    private String imageUrl;
    private LocalDateTime createdAt;
    private Long userId; // Author
    private String userName;
    private String userProfilePic;
    private int likeCount=0;          // total likes
    private int commentCount=0;       // total comments
    private boolean likedByUser;    // did the current user like this post
    
    
    private List<UserDTO> likedBy = new ArrayList<>();        // list of users who liked
    private List<CommentDTO> comments =new ArrayList<>();    // list of comments with user info
    

    

}

File: EduHubConnect-AndroidFrontEnd/app/src/main/java/com/connect/eduhubconnect/dto/PostDTO.kt
package com.connect.eduhubconnect.dto

data class PostDTO(
    val id: Long,
    val content: String,
    val imageUrl: String?,
    val createdAt: String,
    val userId: Long,
    val userName:String,

    var likeCount: Int,
    val commentCount: Int,
    var likedByUser: Boolean,

    val likedBy: List<UserDTO>,
    val comments: List<CommentDTO>
)
